sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -146,73 +143,89 @@ specified explicitly.
 
 [NOTE]
 ====
-If a `work_group_static` object is declared at function scope, the
+If a `work_group_static` object is declared at a block scope, the
 device local memory associated with the object will be identical for all
 usages of that function within the kernel. In cases where a function is called
 multiple times, developers must take care to avoid race conditions (e.g., by
 calling `group_barrier` before and after using the memory).
 ====
 
+Change to SYCL 2020 section `5.9.2 Common address space deduction rules`:
+Namespace scope: if the variable is `work_group_static` object,
+then the variable is assigned to the local address space.
+Otherwise normal rules applies.
+
 SYCL 2020 requires that all global variables accessed by a device function are
 `const` or `constexpr`. This extension lifts that restriction for
 `work_group_static` variables.
 
-[NOTE]
-====
-Since `work_group_static` acts as a view, wrapping an underlying pointer, a
-developer may still choose to declare variables as `const`.
-====
-
 When `T` is a class type or bounded array, the size of the allocation is known
-at compile-time, and a SYCL implementation may embed the size of the allocation
+at compile-time, and a SYCL implementation embeds the size of the allocation
 directly within a kernel. Each instance of `work_group_static<T>` is associated
 with a unique allocation in device local memory.
```

Comment:
The first sentence in this paragraph is no longer needed.  All uses of `work_group_static` have a size that is known at compile time, now that you have removed the possibility of unbounded arrays.